### DISCUSSION TITLE
Domain of variable

### DISCUSSION THREAD

**Discussion by samuel-casasdray** [ERGO-Code/HiGHS#1734] [Q&A] [answered]
I'm using HiGHs with python and I have some issues about the domain of the variable. Can we specify that we only want a solution with natural number or even binary number ?

**Comment by jajhall** [comment, marked as answer]
Yes, the type of a variable can be set to `highspy.HighsVarType.kInteger` and then its bounds used to define the range of integer values that it may take. How you do this in practice depends on how you are defining your model. 

- If you are populating an instance of the `HighsLp` class, then set the entries of the `integrality_` vector to the value `highspy.HighsVarType.kInteger`, and the values of the `colLower_` and `colUpper_` vectors to the desired bounds.
- If you have built the model by defining variables and constraints, you call the following `highspy` method, where `n` is the number of variables that you want to be integer-valued, `indices` is the list of length `n` of the indices of variables to be integer-valued, and `integrality` is a list of length `n`, all of whose entries have the value `highspy.HighsVarType.kInteger`. The definition might be cumbersome if you want all your variables to be integer valued, but it allows for simplicity in terms of number of `highspy` methods and generality of use

   changeColsIntegrality(n, indices, integrality)

Bounds on variables can be changed using the following `highspy` method, `lower` and `upper` are lists of bound values of length `n`

   changeColsBounds(n, indices, lower, upper)

Feel free to ask further questions

**Reply by samuel-casasdray** [reply]
That's perfect, thanks a lot!
And so I have another question, non-related. Do you know if it's possible to print my model in LP format, in the terminal or in a file, I don't care, I just want to see if I create it well?

**Reply by jajhall** [reply]
Yes, call `h.writeModel("foo.lp")`. If you call `h.writeModel("")` you'll get the model to the screen in a human-readable format

**Reply by samuel-casasdray** [reply]
That's perfect again, thanks!
